### ISSUE TITLE
Fix build

### ISSUE THREAD

**Issue by njam** [cargomedia/puppet-packages#797]
Depends on #829 

According to Jenkins the last successful build was 2014-10-16.
We should fix the problems, and make sure the build stays "green".

@kris-lab @ppp0 wdyt?

**Comment by kris-lab** [comment]
@njm is it already done?

**Comment by njam** [comment]
@kris-lab you're asking whether I worked on this already? -> no

**Comment by tomaszdurka** [comment]
Latest build failures:
```
Failures:

  1) Package "puppetdb" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Puppet command failed: Command output contains error: `Error: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install puppetdb' returned 100: Reading package lists...`
     # ./spec/spec_helper.rb:75:in `abort'
     # ./spec/spec_helper.rb:75:in `rescue in block (3 levels) in <top (required)>'
     # ./spec/spec_helper.rb:66:in `block (3 levels) in <top (required)>'
     # ./spec/spec_helper.rb:60:in `each'
     # ./spec/spec_helper.rb:60:in `block (2 levels) in <top (required)>'

  2) Command "timeout --signal=9 30 bash -c "while ! (grep -q 'PuppetDB version' /var/log/puppetdb/puppetdb.log); do sleep 0.5; done"" should return exit status 0
     Failure/Error: it { should return_exit_status 0 }
       expected Command "timeout --signal=9 30 bash -c "while ! (grep -q 'PuppetDB version' /var/log/puppetdb/puppetdb.log); do sleep 0.5; done"" to return exit status 0
     # ./modules/puppet/spec/db/spec.rb:9:in `block (2 levels) in <top (required)>'

  3) Port "8080" should be listening
     Failure/Error: it { should be_listening }
       expected Port "8080" to be listening
     # ./modules/puppet/spec/db/spec.rb:13:in `block (2 levels) in <top (required)>'

  4) Port "8081" should be listening
     Failure/Error: it { should be_listening }
       expected Port "8081" to be listening
     # ./modules/puppet/spec/db/spec.rb:17:in `block (2 levels) in <top (required)>'

  5) Command "s3export" stderr should match "[options] <command> [arguments]"
     Failure/Error: its(:stderr) { should match '[options] <command> [arguments]' }
       expected "" to match "[options] <command> [arguments]"
     # ./modules/s3export_backup/spec/init/spec.rb:5:in `block (2 levels) in <top (required)>'

  6) Package "virtualbox-4.3" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Puppet command failed: Command output contains error: `Error: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install virtualbox-4.3' returned 100: Reading package lists...`
     # ./spec/spec_helper.rb:75:in `abort'
     # ./spec/spec_helper.rb:75:in `rescue in block (3 levels) in <top (required)>'
     # ./spec/spec_helper.rb:66:in `block (3 levels) in <top (required)>'
     # ./spec/spec_helper.rb:60:in `each'
     # ./spec/spec_helper.rb:60:in `block (2 levels) in <top (required)>'

Finished in 137 minutes 38 seconds (files took 0.3333 seconds to load)
556 examples, 6 failures

Failed examples:

rspec ./modules/puppet/spec/db/spec.rb:4 # Package "puppetdb" 
rspec ./modules/puppet/spec/db/spec.rb:9 # Command "timeout --signal=9 30 bash -c "while ! (grep -q 'PuppetDB version' /var/log/puppetdb/puppetdb.log); do sleep 0.5; done"" should return exit status 0
rspec ./modules/puppet/spec/db/spec.rb:13 # Port "8080" should be listening
rspec ./modules/puppet/spec/db/spec.rb:17 # Port "8081" should be listening
rspec ./modules/s3export_backup/spec/init/spec.rb:5 # Command "s3export" stderr should match "[options] <command> [arguments]"
rspec ./modules/virtualbox/spec/default/spec.rb:4 # Package "virtualbox-4.3" 
/usr/bin/ruby1.9.1 -I/var/lib/jenkins/workspace/cargomedia-puppet-packages/.bundle/ruby/1.9.1/gems/rspec-core-3.1.7/lib:/var/lib/jenkins/workspace/cargomedia-puppet-packages/.bundle/ruby/1.9.1/gems/rspec-support-3.1.2/lib /var/lib/jenkins/workspace/cargomedia-puppet-packages/.bundle/ruby/1.9.1/gems/rspec-core-3.1.7/exe/rspec --pattern modules/\*/spec/\*/spec.rb failed
vagrant halt --force
==> wheezy: Forcing shutdown of VM...
Build step 'Execute shell' marked build as failure
Sending e-mails to: sysadmin@cargomedia.ch
Finished: FAILURE
```

**Comment by tomaszdurka** [comment]
Depends on #829

**Comment by tomaszdurka** [comment]
- `test::puppet::db` in https://github.com/cargomedia/puppet-packages/pull/852
- `test:s3export_backup` fixed
- `test:s3export_backup` seems to not pop anymore (maybe just a bit flaky)

**Comment by tomaszdurka** [comment]
Today tests brought these errors:
```
Failures:

  1) bower Package "bower" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: Could not update: Execution of '/usr/bin/npm install --global bower@1.3.12' returned 1: npm http GET https://registry.npmjs.org/bower`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  2) bower Command "bower --version" exit_status 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: Could not update: Execution of '/usr/bin/npm install --global bower@1.3.12' returned 1: npm http GET https://registry.npmjs.org/bower`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  3) puppet::master puppetfile File "/etc/puppet/modules/mysql/metadata.json" content 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: /Stage[main]/Puppet::Master::Puppetfile/Exec[librarian update and rsync]: Failed to call refresh: cd /etc/puppet && librarian-puppet update && /usr/local/bin/sync_hiera.sh returned 1 instead of one of [0]`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  4) puppet::master puppetfile File "/foobar" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: /Stage[main]/Puppet::Master::Puppetfile/Exec[librarian update and rsync]: Failed to call refresh: cd /etc/puppet && librarian-puppet update && /usr/local/bin/sync_hiera.sh returned 1 instead of one of [0]`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  5) puppet::master puppetfile File "/usr/local/bin/sync_hiera.sh" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: /Stage[main]/Puppet::Master::Puppetfile/Exec[librarian update and rsync]: Failed to call refresh: cd /etc/puppet && librarian-puppet update && /usr/local/bin/sync_hiera.sh returned 1 instead of one of [0]`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  6) puppet::master puppetfile File "/usr/local/bin/sync_hiera.sh" content 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: /Stage[main]/Puppet::Master::Puppetfile/Exec[librarian update and rsync]: Failed to call refresh: cd /etc/puppet && librarian-puppet update && /usr/local/bin/sync_hiera.sh returned 1 instead of one of [0]`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  7) puppet::master puppetfile Cron 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: /Stage[main]/Puppet::Master::Puppetfile/Exec[librarian update and rsync]: Failed to call refresh: cd /etc/puppet && librarian-puppet update && /usr/local/bin/sync_hiera.sh returned 1 instead of one of [0]`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'
```

Not sure if this can ever succeed with so random http failures.

**Comment by ppp0** [comment]
I also had the feeling that errors are not consistent, unfortunately - except that it seems to be `http` problems arising mostly when doing some sort of package install/update.. Not sure if we should give another proxy a try (instead of `pòlipo`).. or have the failed test repeat at a later stage? May be after resetting `.proxy-cache` and/or the whole vagrant box?

**Comment by tomaszdurka** [comment]
Well there were tests which failed always or quite often on ci. I think we solved all of them now so hopefully we can get a green build some day or even often :panda_face: 

@ppp0 @njam wdyt?

**Comment by tomaszdurka** [comment]
As said let's keep this issue open for couple more days and observe builds.
I will post outputs everyday to see what kind of failures - if any we have.

**Comment by njam** [comment]
:+1:

**Comment by tomaszdurka** [comment]
Doh! 
19th Feb:
```
Failures:

  1) deb_multimedia Package "deb-multimedia-keyring" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: apt-key adv --keyserver pgp.mit.edu --recv-keys 65558117 returned 2 instead of one of [0]`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  2) ffmpeg File "/usr/local/bin/ffmpeg" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: apt-key adv --keyserver pgp.mit.edu --recv-keys 65558117 returned 2 instead of one of [0]`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  3) nfs File "/tmp/source/foo" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: /Stage[main]/Main/Node[default]/Nfs::Mount[/tmp/mounted]/Mount::Entry[/tmp/mounted]/Exec[/usr/sbin/mount-check.sh /tmp/mounted]: Failed to call refresh: /usr/sbin/mount-check.sh /tmp/mounted returned 1 instead of one of [0]`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  4) nfs Service "nfs-common" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: /Stage[main]/Main/Node[default]/Nfs::Mount[/tmp/mounted]/Mount::Entry[/tmp/mounted]/Exec[/usr/sbin/mount-check.sh /tmp/mounted]: Failed to call refresh: /usr/sbin/mount-check.sh /tmp/mounted returned 1 instead of one of [0]`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  5) nfs Service "nfs-common" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: /Stage[main]/Main/Node[default]/Nfs::Mount[/tmp/mounted]/Mount::Entry[/tmp/mounted]/Exec[/usr/sbin/mount-check.sh /tmp/mounted]: Failed to call refresh: /usr/sbin/mount-check.sh /tmp/mounted returned 1 instead of one of [0]`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  6) nfs Service "nfs-kernel-server" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: /Stage[main]/Main/Node[default]/Nfs::Mount[/tmp/mounted]/Mount::Entry[/tmp/mounted]/Exec[/usr/sbin/mount-check.sh /tmp/mounted]: Failed to call refresh: /usr/sbin/mount-check.sh /tmp/mounted returned 1 instead of one of [0]`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  7) nfs Service "nfs-kernel-server" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: /Stage[main]/Main/Node[default]/Nfs::Mount[/tmp/mounted]/Mount::Entry[/tmp/mounted]/Exec[/usr/sbin/mount-check.sh /tmp/mounted]: Failed to call refresh: /usr/sbin/mount-check.sh /tmp/mounted returned 1 instead of one of [0]`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  8) nfs Command "monit summary | grep nfs-server" exit_status 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: /Stage[main]/Main/Node[default]/Nfs::Mount[/tmp/mounted]/Mount::Entry[/tmp/mounted]/Exec[/usr/sbin/mount-check.sh /tmp/mounted]: Failed to call refresh: /usr/sbin/mount-check.sh /tmp/mounted returned 1 instead of one of [0]`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  9) nfs Command "ps ax | grep "\[nfsd\]" | wc -l" exit_status 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: /Stage[main]/Main/Node[default]/Nfs::Mount[/tmp/mounted]/Mount::Entry[/tmp/mounted]/Exec[/usr/sbin/mount-check.sh /tmp/mounted]: Failed to call refresh: /usr/sbin/mount-check.sh /tmp/mounted returned 1 instead of one of [0]`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  10) nfs Command "ps ax | grep "\[nfsd\]" | wc -l" stdout 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: /Stage[main]/Main/Node[default]/Nfs::Mount[/tmp/mounted]/Mount::Entry[/tmp/mounted]/Exec[/usr/sbin/mount-check.sh /tmp/mounted]: Failed to call refresh: /usr/sbin/mount-check.sh /tmp/mounted returned 1 instead of one of [0]`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  11) nfs File "/tmp/source/bar" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: /Stage[main]/Main/Node[default]/Nfs::Mount[/tmp/mounted]/Mount::Entry[/tmp/mounted]/Exec[/usr/sbin/mount-check.sh /tmp/mounted]: Failed to call refresh: /usr/sbin/mount-check.sh /tmp/mounted returned 1 instead of one of [0]`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  12) nfs File "/nfsexport/shared/foo" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: /Stage[main]/Main/Node[default]/Nfs::Mount[/tmp/mounted]/Mount::Entry[/tmp/mounted]/Exec[/usr/sbin/mount-check.sh /tmp/mounted]: Failed to call refresh: /usr/sbin/mount-check.sh /tmp/mounted returned 1 instead of one of [0]`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  13) nfs File "/nfsexport/shared/bar" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: /Stage[main]/Main/Node[default]/Nfs::Mount[/tmp/mounted]/Mount::Entry[/tmp/mounted]/Exec[/usr/sbin/mount-check.sh /tmp/mounted]: Failed to call refresh: /usr/sbin/mount-check.sh /tmp/mounted returned 1 instead of one of [0]`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  14) nfs File "/tmp/mounted" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: /Stage[main]/Main/Node[default]/Nfs::Mount[/tmp/mounted]/Mount::Entry[/tmp/mounted]/Exec[/usr/sbin/mount-check.sh /tmp/mounted]: Failed to call refresh: /usr/sbin/mount-check.sh /tmp/mounted returned 1 instead of one of [0]`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  15) nfs File "/tmp/mounted" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: /Stage[main]/Main/Node[default]/Nfs::Mount[/tmp/mounted]/Mount::Entry[/tmp/mounted]/Exec[/usr/sbin/mount-check.sh /tmp/mounted]: Failed to call refresh: /usr/sbin/mount-check.sh /tmp/mounted returned 1 instead of one of [0]`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  16) nfs File "/tmp/mounted" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: /Stage[main]/Main/Node[default]/Nfs::Mount[/tmp/mounted]/Mount::Entry[/tmp/mounted]/Exec[/usr/sbin/mount-check.sh /tmp/mounted]: Failed to call refresh: /usr/sbin/mount-check.sh /tmp/mounted returned 1 instead of one of [0]`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  17) nfs File "/tmp/mounted" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: /Stage[main]/Main/Node[default]/Nfs::Mount[/tmp/mounted]/Mount::Entry[/tmp/mounted]/Exec[/usr/sbin/mount-check.sh /tmp/mounted]: Failed to call refresh: /usr/sbin/mount-check.sh /tmp/mounted returned 1 instead of one of [0]`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  18) nfs File "/tmp/mounted" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: /Stage[main]/Main/Node[default]/Nfs::Mount[/tmp/mounted]/Mount::Entry[/tmp/mounted]/Exec[/usr/sbin/mount-check.sh /tmp/mounted]: Failed to call refresh: /usr/sbin/mount-check.sh /tmp/mounted returned 1 instead of one of [0]`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  19) nfs File "/tmp/mounted/bar" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: /Stage[main]/Main/Node[default]/Nfs::Mount[/tmp/mounted]/Mount::Entry[/tmp/mounted]/Exec[/usr/sbin/mount-check.sh /tmp/mounted]: Failed to call refresh: /usr/sbin/mount-check.sh /tmp/mounted returned 1 instead of one of [0]`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  20) nfs File "/nfsexport/shared" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: /Stage[main]/Main/Node[default]/Nfs::Mount[/tmp/mounted]/Mount::Entry[/tmp/mounted]/Exec[/usr/sbin/mount-check.sh /tmp/mounted]: Failed to call refresh: /usr/sbin/mount-check.sh /tmp/mounted returned 1 instead of one of [0]`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  21) nfs File "/nfsexport/shared" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: /Stage[main]/Main/Node[default]/Nfs::Mount[/tmp/mounted]/Mount::Entry[/tmp/mounted]/Exec[/usr/sbin/mount-check.sh /tmp/mounted]: Failed to call refresh: /usr/sbin/mount-check.sh /tmp/mounted returned 1 instead of one of [0]`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  22) nfs File "/nfsexport/shared" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: /Stage[main]/Main/Node[default]/Nfs::Mount[/tmp/mounted]/Mount::Entry[/tmp/mounted]/Exec[/usr/sbin/mount-check.sh /tmp/mounted]: Failed to call refresh: /usr/sbin/mount-check.sh /tmp/mounted returned 1 instead of one of [0]`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  23) nfs File "/nfsexport/shared" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: /Stage[main]/Main/Node[default]/Nfs::Mount[/tmp/mounted]/Mount::Entry[/tmp/mounted]/Exec[/usr/sbin/mount-check.sh /tmp/mounted]: Failed to call refresh: /usr/sbin/mount-check.sh /tmp/mounted returned 1 instead of one of [0]`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  24) puppet::db Package "puppetdb" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install puppetdb' returned 100: Reading package lists...`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  25) puppet::db Service "puppetdb" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install puppetdb' returned 100: Reading package lists...`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  26) puppet::db Service "puppetdb" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install puppetdb' returned 100: Reading package lists...`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  27) puppet::db Command "timeout --signal=9 30 bash -c "while ! (netstat -altp |grep -q 'java'); do sleep 0.5; done"" exit_status 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install puppetdb' returned 100: Reading package lists...`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  28) puppet::db Port "8080" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install puppetdb' returned 100: Reading package lists...`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  29) puppet::db Port "8081" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install puppetdb' returned 100: Reading package lists...`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

Finished in 140 minutes 24 seconds (files took 0.39767 seconds to load)
572 examples, 29 failures
```

**Comment by tomaszdurka** [comment]
20th Feb:
```
Failures:

  1) puppet::db Package "puppetdb" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install puppetdb' returned 100: Reading package lists...`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  2) puppet::db Service "puppetdb" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install puppetdb' returned 100: Reading package lists...`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  3) puppet::db Service "puppetdb" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install puppetdb' returned 100: Reading package lists...`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  4) puppet::db Command "timeout --signal=9 30 bash -c "while ! (netstat -altp |grep -q 'java'); do sleep 0.5; done"" exit_status 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install puppetdb' returned 100: Reading package lists...`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  5) puppet::db Port "8080" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install puppetdb' returned 100: Reading package lists...`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

  6) puppet::db Port "8081" 
     Failure/Error: Unable to find matching line from backtrace
     SystemExit:
       Command failed: Puppet command failed: `Error: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install puppetdb' returned 100: Reading package lists...`
       
     # ./spec/lib/puppet_spec.rb:64:in `abort'
     # ./spec/lib/puppet_spec.rb:64:in `rescue in block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:54:in `block in apply_manifests'
     # ./spec/lib/puppet_spec.rb:46:in `each'
     # ./spec/lib/puppet_spec.rb:46:in `apply_manifests'
     # ./spec/spec_helper.rb:24:in `block (2 levels) in <top (required)>'

Finished in 150 minutes 1 second (files took 0.39971 seconds to load)
581 examples, 6 failures
```

**Comment by tomaszdurka** [comment]
Depends on #869

**Comment by tomaszdurka** [comment]
Hopefully fixed by #869.

**Comment by tomaszdurka** [comment]
Yay worked!
http://ci.cargomedia.ch:8080/job/cargomedia-puppet-packages/203/

**Comment by njam** [comment]
:star2:

**Comment by ppp0** [comment]
Shall we add this proxy bypass also to other failing specs?

I see virtualbox is failing in a quite similar way, for instance (see http://ci.cargomedia.ch:8080/job/cargomedia-puppet-packages/205/console)
